lib/simplifyExpression/collectAndCombineSearch/multiplyLikeTerms.js

```diff
@@ -93,51 +87,82 @@ function multiplyPolynomialTerms(node) {
 function addOneExponent(node) {
   const newNode = clone(node);
   let change = false;
-
   let changeGroup = 1;
-  newNode.args.forEach((child, i) => {
-    const polyTerm = new Node.PolynomialTerm(child);
-    if (!polyTerm.getExponentNode())  {
-      newNode.args[i] = Node.Creator.polynomialTerm(
-        polyTerm.getSymbolNode(),
-        Node.Creator.constant(1),
-        polyTerm.getCoeffNode());
-
-      newNode.args[i].changeGroup = changeGroup;
-      node.args[i].changeGroup = changeGroup; // note that this is the "oldNode"
-
-      change = true;
-      changeGroup++;
-    }
-  });
+  if (checks.canMultiplyLikeTermConstantNodes(node)) {
+    newNode.args.forEach((child, i) => {
+      const constTerm = new Node.ConstantTerms(child);
+      if (!constTerm.getExponentNode()) {
+        newNode.args[i] = Node.Creator.constantTerm(
+          constTerm.getBaseNode(),
+          Node.Creator.constant(1),
+          constTerm.getCoeffNode());
+
+        newNode.args[i].changeGroup = changeGroup;
+        node.args[i].changeGroup = changeGroup; // note that this is the "oldNode"
+
+        change = true;
+        changeGroup++;
+      }
+    });
+  }
+  else {
+    newNode.args.forEach((child, i) => {
+      const polyTerm = new Node.PolynomialTerm(child);
+      if (!polyTerm.getExponentNode()) {
+        newNode.args[i] = Node.Creator.polynomialTerm(
+          polyTerm.getSymbolNode(),
+          Node.Creator.constant(1),
+          polyTerm.getCoeffNode());
+
+        newNode.args[i].changeGroup = changeGroup;
+        node.args[i].changeGroup = changeGroup; // note that this is the "oldNode"
+
+        change = true;
+        changeGroup++;
+      }
+    });
+  }
 
   if (change) {
     return Node.Status.nodeChanged(
-        ChangeTypes.ADD_EXPONENT_OF_ONE, node, newNode, false);
+      ChangeTypes.ADD_EXPONENT_OF_ONE, node, newNode, false);
   }
   else {
     return Node.Status.noChange(node);
   }
 }
-
 // Given a product of polynomial terms, groups the exponents into a sum
```

Comment:
Hmm, been thinking about this some more. If we add those two functions then the call for them in the code above will be more unclear, so it still might be better to keep as it is. That is to have one function with two cases.